recipes/pysdl2/locate_sdl.patch

```diff
@@ -0,0 +1,18 @@
+--- sdl2/__init__.py
++++ sdl2/__init__.py
+@@ -1,4 +1,15 @@
+ """SDL2 wrapper package"""
++# Direct this module to Ana/Miniconda's lib folder, where the SDL2 libraries
++# can be found.
++import os
++
++if os.name == "posix":
++	os.environ["PYSDL2_DLL_PATH"] = os.path.abspath(os.path.join(
++		os.path.dirname(__file__), os.pardir, os.pardir, os.pardir))
++elif os.name == "nt":
++	os.environ["PYSDL2_DLL_PATH"] = os.path.abspath(os.path.join(
++		os.path.dirname(__file__), os.pardir, os.pardir, os.pardir, "Library", "bin"))
```

Comment:
Thanks for looking into this. Your suggestion requires changing 4 or 5 files (dll.py, sdlgfx.py sdlimage.py, etc.) so I think its best to still patch this in the __init__.py and set the environment variable. I have now changed it in such a way that it only sets the environment variable if it is not already present:

~~~
import os
if not "PYSDL2_DLL_PATH" in os.environ:
	if os.name == "posix":
		os.environ["PYSDL2_DLL_PATH"] = os.path.abspath(os.path.join(
			os.path.dirname(__file__), os.pardir, os.pardir, os.pardir))
	elif os.name == "nt":
		os.environ["PYSDL2_DLL_PATH"] = os.path.abspath(os.path.join(
			os.path.dirname(__file__), os.pardir, os.pardir, os.pardir, "Library", "bin"))
~~~

Even though I agree with you that this is a hacky solution, I still don't see the harm in always letting conda set this var to be honest, and I predict much more trouble if a potential user's setting is honored first. If I'm not mistaken, it is the aim to let conda always use its own libraries, and rely as little as possible on modules outside the conda environment. 
In the current setting, if a user for instance has set PYSDL2_DLL_PATH to c:\src in its OS's standard environment vars for a different Python installation, then this setting will always be used in conda too, even though it should use its own libraries. 
I furthermore doubt the user will ever change environment var if the sdl2 libraries are correctly found. The whole PYSDL2_DLL_PATH environment variable is to give pysdl2 a helping hand in location the sdl2 libraries, which is exactly what is being done with this patch.
Just my 2 cents ;)